### ISSUE TITLE
Docker extension doesn't update files from the container

### ISSUE THREAD

**Issue by practicasAlberto** [microsoft/vscode-docker#3923]
I'm running an image with "docker run -v [local route]:[container route]".

The mount works correctly, when I modify the local files they get modified inside the container, but if I look the files with the docker extension after doing a local modify, the docker extension shows me the file before the change.

I tried to refresh the containers section several times but the extension doesn't show me the change.

When I make a change in a file inside the container using the exension then my local file gets modified inmediatly.

In conclusion the problem is that the extension doesn't show the files updated correctly.

**Comment by bwateratmsft** [comment]
So I've found what I believe is happening here. Because our answer to VSCode's `stat` calls is [fake](https://github.com/microsoft/vscode-docker/blob/53ab512ea83694da03611a54263ebd969c93caa6/src/runtimes/files/ContainerFilesProvider.ts#L36-L45) (the performance of rapid `stat` calls into a container is atrocious), VSCode doesn't know that the file has changed, and always shows the same cached value. The download command directly streams the file so download will always fetch the latest file.

To fix this would require doing real `stat` calls, which VSCode sometimes fires _very_ rapidly. This would result in high CPU usage, poor performance, etc.

A potential workaround would be to cache the real `stat` info at the time of clicking Refresh. If that were done, the file would not update until the refresh happened, but it would update. However, this could be a significant amount of work to get right.

We discussed and have decided not to fix this issue unless we hear more feedback about it, due to the difficulty to fix vs. the prevalence of users encountering the problem.
